README.md

```diff
@@ -2,27 +2,87 @@
 
 a set of reusable utilities to allow Wagtail implementors to easily generate data migrations for changes to StreamField block structure
 
-
 [![License](https://img.shields.io/badge/License-BSD_3--Clause-blue.svg)](https://opensource.org/licenses/BSD-3-Clause)
 
 [![PyPI version](https://badge.fury.io/py/streamfield-migration-toolkit.svg)](https://badge.fury.io/py/streamfield-migration-toolkit)
 [![Wagtail Hallo CI](https://github.com/wagtail/streamfield-migration-toolkit/actions/workflows/test.yml/badge.svg)](https://github.com/wagtail/streamfield-migration-toolkit/actions/workflows/test.yml)
 
-## Links
+# Contents
 
-- [Documentation](https://github.com/wagtail/streamfield-migration-toolkit/blob/main/README.md)
-- [Changelog](https://github.com/wagtail/streamfield-migration-toolkit/blob/main/CHANGELOG.md)
-- [Contributing](https://github.com/wagtail/streamfield-migration-toolkit/blob/main/CHANGELOG.md)
-- [Discussions](https://github.com/wagtail/streamfield-migration-toolkit/discussions)
-- [Security](https://github.com/wagtail/streamfield-migration-toolkit/security)
+- [Introduction](#introduction)
+  - [Why data migrations?](#why-data-migrations)
+- [Quick Start](#quick-start)
+- [Reference](docs/REFERENCE.md)
+- [Usage](docs/USAGE.md)
+- [Contributing](docs/CONTRIBUTING.md)
 
-## Supported versions
+# Introduction
+
+This package aims to make it easier for developers using StreamField who need to write data
+migrations when making changes involving blocks/block structure in the StreamField. We expose a
+set of utilities for commonly made changes such as renaming or removing blocks, as well as utility
+functions for recursing through existing Streamfield data and applying changes, which makes it
+easier to create custom logic for applying changes too.
+
+## Why Data Migrations?
+
+A streamfield is stored as a single column of JSON data in the database, where there are blocks stored (which can be nested to form complex block structures) within the JSON representation. However, as far as django is concerned when making schema migrations, everything inside this column is just a string of JSON data and the schema doesn’t change regardless of the content/structure of the StreamField since it is the same field type before and after the change. Therefore whenever changes are made to StreamFields, any existing data must be changed into the required structure by using a data migration created manually by the user.
```

Comment:
Maybe just: "A StreamField is stored as a single column of JSON data in the database. Blocks are stored as structures within the JSON, and can be nested` - to avoid the slightly convoluted first sentence